graph-query-ir/src/main/java/oracle/pgql/lang/ResultSet.java

```diff
@@ -3,14 +3,290 @@
+   *
+   * @return <code>true</code> if the new current row is valid; <code>false</code> if the new cursor is positioned
+   * before the first row
+   */
+  public boolean previous() throws PgqlException;
+
+  /**
+   * Place the cursor before the first row.
+   */
+  public void beforeFirst() throws PgqlException;
+
+  /**
+   * Place the cursor after the last row.
+   */
+  public void afterLast() throws PgqlException;
+
+  /**
+   * Place the cursor to the given row number in this ResultSet object.
+   *
+   * @return <code>true</code> if the cursor is moved to a position in the ResultSet object; <code>false</code> if
+   * the cursor is moved before the first or after the last row
+   * @throws PgqlException if a connection error occurs or when this method is called on a closed result set
+   */
+  public boolean absolute(int row) throws PgqlException;
+
+  /**
+   * Moves the cursor a relative number of row with repect to the current position. A negative number will move the
+   * cursor backwards.
+   *
+   * @return <code>true</code> if the cursor is moved to a position in the ResultSet object; <code>false</code> if
+   * the cursor is moved before the first or after the last row
+   */
+  public boolean relative(int rows) throws PgqlException;
+
+  /**
+   * Gets the value of the designated element by element name as an {@link Object}
+   *
+   * @param elementIdx element index
+   * @return {@link Object}
+   */
+  public Object getObject(int elementIdx) throws PgqlException;
+
+  /**
+   * Gets the value of the designated element by element name as an {@link Object}
+   *
+   * @param elementName element name
+   * @return {@link Object}
+   */
+  public Object getObject(String elementName) throws PgqlException;
+
+  /**
+   * Gets the value of the designated element by element index as a {@link String}
+   *
+   * @param elementIdx element index
+   * @return {@link String}
+   */
+  public String getString(int elementIdx) throws PgqlException;
+
+  /**
+   * Gets the value of the designated element by element name as a {@link String}
+   *
+   * @param elementName element name
+   * @return {@link String}
+   */
+  public String getString(String elementName) throws PgqlException;
+
+  /**
+   * Gets the value of the designated element by element index as an {@link Integer}
+   *
+   * @param elementIdx element index
+   * @return {@link Integer}
+   */
+  public Integer getInteger(int elementIdx) throws PgqlException;
+
+  /**
+   * Gets the value of the designated element by element name as an {@link Integer}
+   *
+   * @param elementName element name
+   * @return {@link Integer}
+   */
+  public Integer getInteger(String elementName) throws PgqlException;
+
+  /**
+   * Gets the value of the designated element by element index as a {@link Long}
+   *
+   * @param elementIdx element index
+   * @return {@link Long}
+   */
+  public Long getLong(int elementIdx) throws PgqlException;
```

Comment:
unless we want to be able to return `null`... there is a use case for it. Let's ask Jinha if this was a design decision and if he thinks it should be boxed.